src/App.tsx

```diff
@@ -1,19 +1,124 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useContext, useEffect, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './TodoList/Todolist';
+import { Footer } from './Footer/Footer';
+import { ErrorNotification } from './ErrorNotification/ErrorNotification';
+import { getTodos, createTodo, deleteTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setfilterType] = useState('all');
+  const [error, setError] = useState<string | null>('');
+  const [title, setTitle] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+
+  if (error) {
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
+  }
+
+  const filteredTodos = todos
+    .filter(todo => {
+      switch (filterType) {
+        case FilterType.Active:
+          return !todo.completed;
+        case FilterType.Completed:
+          return todo.completed;
+        case FilterType.All:
+          return todo;
+        default:
+          return 0;
+      }
+    });
+
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user.id;
+  }
+
+  userId = 0;
+
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    getTodos(userId)
+      .then(setTodos)
+      .catch(() => (setError('Unable to load todo from server')));
   }, []);
 
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (title.trim().length === 0) {
+      setError('Title can\'t be empty');
+      setTitle('');
+
+      return;
+    }
+
+    if (!user) {
+      return;
+    }
+
+    const newTodoAdd = {
+      id: 0,
+      userId: user.id,
+      title,
+      completed: false,
+    };
+
+    setIsAdding(true);
+
+    setTodos([...todos, newTodoAdd]);
+
+    try {
+      const newTodo = await createTodo(title, userId);
+
+      setTodos([...todos, newTodo]);
+    } catch {
+      setError('Unable to add a todo');
+      setTodos(filteredTodos.filter(todo => todo.id !== 0));
+    }
```

Comment:
On my opinion, this code is more easy `createTodo()
.then()
.catch()
.finally();`